pkg/controllers/user/cis/rbac.go

```diff
@@ -0,0 +1,134 @@
+package cis
+
+import (
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/namespace"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	rbacv1 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	cisReadOnlyRoleName = "cis-readonly"
+)
+
+type cisScanRBACHandler struct {
+	roles             rbacv1.RoleInterface
+	roleLister        rbacv1.RoleLister
+	roleBindings      rbacv1.RoleBindingInterface
+	roleBindingLister rbacv1.RoleBindingLister
+}
+
+func (rh *cisScanRBACHandler) Sync(key string, crtb *v3.ClusterRoleTemplateBinding) (runtime.Object, error) {
+	if crtb == nil {
+		return nil, nil
+	}
+	if crtb.DeletionTimestamp != nil {
+		return rh.handleDelete(key, crtb)
```

Comment:
We like to avoid finalizers as much as we can, but for RBAC, its important to guarantee that permissions are correct. You probably need to convert this to a lifecycle so that you have a finalizer and an explicit delete function (or tack it into an existing CRTB lifecycle.